extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/ConfigPropertiesUtil.java

```diff
@@ -0,0 +1,301 @@
+package io.quarkus.arc.deployment.configproperties;
+
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+
+import org.eclipse.microprofile.config.Config;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.ParameterizedType;
+import org.jboss.jandex.Type;
+
+import io.quarkus.gizmo.BranchResult;
+import io.quarkus.gizmo.BytecodeCreator;
+import io.quarkus.gizmo.ClassCreator;
+import io.quarkus.gizmo.ClassOutput;
+import io.quarkus.gizmo.MethodCreator;
+import io.quarkus.gizmo.MethodDescriptor;
+import io.quarkus.gizmo.ResultHandle;
+import io.smallrye.config.SmallRyeConfig;
+import io.smallrye.config.StringUtil;
+
+final class ConfigPropertiesUtil {
+
+    static final String PACKAGE_TO_PLACE_GENERATED_CLASSES = "io.quarkus.arc.runtime.config";
+
+    private ConfigPropertiesUtil() {
+    }
+
+    /**
+     * Generates a class like:
+     * 
+     * <pre>
+     * final class AbstractConfigPropertiesHelper {
+     *
+     *     void collectionConfigProperty(Config config, String propertyValue,
+     *             Class clazz, Collection collection) {
+     *
+     *         SmallRyeConfig smallRyeConfig = (SmallRyeConfig) config;
+     *         String[] split = StringUtil.split(propertyValue);
```

Comment:
This is much more complex than necessary.  You can instead just use [`newCollectionConverter`](https://github.com/smallrye/smallrye-config/blob/master/implementation/src/main/java/io/smallrye/config/Converters.java#L205-L217) and save yourself some work.

Reply:
Thanks for pointing it out. I'll have a look

Reply:
Done